src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageManagerControl.xaml.cs

```diff
@@ -1132,13 +1133,15 @@ private void Filter_SelectionChanged(object sender, FilterChangedEventArgs e)
                 NuGetUIThreadHelper.JoinableTaskFactory.RunAsync(async () =>
                 {
                     await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+                    var sw = Stopwatch.StartNew();
```

Comment:
Have you validated that the numberrs reported make sense and that they correlate with what you might be seeing on the screen? 

We have a few "RunAsync" tasks, which means they are fire and forget, so if we have nested fire and forget, you won't really capture the time taken for the nested tasks even if they do affect the tab loading.

Reply:
Please check latest change.
I manually test using my phone stopwatch and this numbers, they almost match. For example: one is 29sec, other is 30 sec, mostly like difference happen because my 2 hands are not really in sync when pressing.

Reply:
Unless all 25 packages in the browse tab's search results are either deprecated or have vulnerabilities, then you won't be able to visually see when all the work is complete.

@donnie-msft  is working on almost the exact same issue, you should talk to him to share knowledge.

Reply:
> Unless all 25 packages in the browse tab's search results are either deprecated or have vulnerabilities, then you won't be able to visually see when all the work is complete.
> 
> @donnie-msft is working on almost the exact same issue, you should talk to him to share knowledge.

Yes, we did discuss this https://github.com/NuGet/Client.Engineering/issues/1428#issuecomment-1045485850.
I changed code here, please check latest change.

Reply:
So is this measurement capturing the extra calls made for deprecation/vulnerabilities? 
@zivkan's suggesting no. The discussion you pointed to doesn't really provide an answer.

Reply:
My work is for adhoc benchmarking, but I think your goal is to introduce telemetry that somewhat overlaps with my work.

Particularly this [commit](https://github.com/NuGet/NuGet.Client/commit/e837672effa93a0faee968ba324632f9d439bf6c) should show my approach and where I wrapped events with ETW.